src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -163,6 +163,14 @@ def version(self):
             self._version = self.dist.parsed_version
         return self._version
 
+    def format_for_error(self):
+        # type: () -> str
+        return "{} {} (from {})".format(
+            self.name,
+            self.version,
+            self.link.file_path
```

Comment:
Does this work for remote links?

Reply:
OK, I guess that's another test to add :-)
Thanks.

Reply:
I added a fix, but couldn't work out a reasonable way to add a test. Suggestions welcome :-)